FrontBundle/Controller/NodeController.php

```diff
@@ -77,19 +77,17 @@ protected function updateNodeResponse(Response $response, ReadNodeInterface $nod
     }
 
     /**
+     * Render the node version given by encoded $token
+     * 
+     * @param string  token
      * @param Request $request
      *
-     * @Config\Route("/preview", name="open_orchestra_front_node_preview")
-     * @Config\Method({"GET"})
-     *
      * @return Response
      */
-    public function previewAction(Request $request)
+    public function previewAction($token, Request $request)
     {
-        $token = $request->get('token');
         $decryptedToken = $this->get('open_orchestra_base.manager.encryption')->decrypt($token);
-        $node = $this->get('open_orchestra_model.repository.node')->find($decryptedToken);
-        $this->get('open_orchestra_display.manager.site')->setSiteId($node->getSiteId());
+        $node = $this->get('open_orchestra_model.repository.node')->find(new \MongoId($decryptedToken));
```

Comment:
Do NOT create a MongoId class HERE!!

This will break the independance with mongo odm!

You have to put the find method in the interface, and then create the MongoId class there if needed!